mlflow/models/evaluation/default_evaluator.py

```diff
@@ -693,7 +679,9 @@ def plot_feature_importance():
     def _evaluate_sklearn_model_score_if_scorable(self):
         if self.model_loader_module == "mlflow.sklearn":
             try:
-                score = self.raw_model.score(self.X, self.y)
+                score = self.raw_model.score(
+                    self.X.copy_in_case_of_mutation(), self.y.copy_in_case_of_mutation()
```

Comment:
It might be alright to just make a copy when the model is involved.

Reply:
Addressed by adding a `get_original()` method to `_MutationGuardedDataset`; The benefit of this class is that it requires explicit API calls to obtain either the original dataset or a copy thereof, requiring the developer to think about whether their workflow might mutate the object.